tests/test_kernel_language.py

```diff
@@ -287,13 +285,31 @@ class TestParticle(ptype[mode]):
                        lon=np.linspace(0., 1., npart),
                        lat=np.zeros(npart) + 0.5)
     series = random_series(npart, rngfunc, rngargs, mode)
-    rnglib = 'ParcelsRandom' if mode == 'jit' else 'random'
     kernel = expr_kernel('TestRandom_%s' % rngfunc, pset,
-                         '%s.%s(%s)' % (rnglib, rngfunc, ', '.join([str(a) for a in rngargs])))
+                         'ParcelsRandom.%s(%s)' % (rngfunc, ', '.join([str(a) for a in rngargs])))
     pset.execute(kernel, endtime=1., dt=1.)
     assert np.allclose(pset.p, series, atol=1e-9)
 
 
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('concat', [False, True])
+def test_random_kernel_concat(fieldset, mode, concat):
+    class TestParticle(ptype[mode]):
+        p = Variable('p', dtype=np.float32)
+
+    pset = ParticleSet(fieldset, pclass=TestParticle, lon=0, lat=0)
+
+    def RandomKernel(particle, fieldset, time):
+        particle.p += ParcelsRandom.uniform(0, 1)
+
+    def AddOne(particle, fieldset, time):
+        particle.p += 1.
+
+    kernels = pset.Kernel(RandomKernel)+pset.Kernel(AddOne) if concat else RandomKernel
+    pset.execute(kernels, dt=0)
+    assert pset.p > 1 if concat else pset.p < 1
```

Comment:
that's a nice test making sure that concatenated kernels just the same RNG

Reply:
Thanks!